image_transport/include/image_transport/image_transport.hpp

```diff
@@ -288,6 +289,19 @@ class  ImageTransport
   std::unique_ptr<Impl> impl_;
 };
 
+struct ImageTransport::Impl
+{
+  rclcpp::Node::SharedPtr node_;
+};
+
+template <typename NodeT>
+ImageTransport::ImageTransport(NodeT node)
+  : impl_(std::make_unique<ImageTransport::Impl>())
+{
+  std::shared_ptr<rclcpp::Node> ptr{node};
```

Comment:
This should be a dynamic cast.

Reply:
> This should be a dynamic cast.

I don't see the inheritance for the ImageTransport class. What should the node cast for? Pardon me for not understanding this enough, but could I ask for some advice?

Reply:
Sorry, my comment was incorrect.

Because there is no common base class for nodes, you need to do one of the following:

- Template the entire `ImageTransport` class on the node type so that `imp_->node_` can store the templated node type.
- Instead of receiving a node instance, receive pointers to the interfaces that you need to use only (such as the `NodeTopicsInterface` pointer) and store those instead of `impl_->node_`.